### ISSUE TITLE
Support API INTEGRATION of type git_https_api

### ISSUE THREAD

**Issue by olg200492** [Snowflake-Labs/terraform-provider-snowflake#2772]
### Use Cases or Problem Statement

Add in [CREATE API INTEGRATION](https://docs.snowflake.com/en/sql-reference/sql/create-api-integration) resource support for git_https_api api provider
As the example [git_https_api](https://docs.snowflake.com/en/developer-guide/git/git-setting-up#create-an-api-integration-for-interacting-with-the-repository-api) 

### Category

category:resource

### Object type(s)

resource:api_integration

### How much impact is this issue causing?

Medium

### Would you like to implement a fix?

- [ ] Yeah, I'll take it 😎

**Comment by sfc-gh-asawicki** [comment]
Hey @olg200492. Thanks for reaching out to us.

I will add this as a part of https://github.com/Snowflake-Labs/terraform-provider-snowflake/blob/main/ROADMAP.md#preparing-the-rest-of-the-fundamental-ga-objects-for-the-provider-v1 (also here: https://github.com/Snowflake-Labs/terraform-provider-snowflake/blob/main/v1-preparations/REMAINING_GA_OBJECTS.MD).

We have recently refreshed our contribution guidelines (announced here: https://github.com/Snowflake-Labs/terraform-provider-snowflake/discussions/2756) and we will welcome a contribution.

For now, you can use https://registry.terraform.io/providers/Snowflake-Labs/snowflake/latest/docs/resources/unsafe_execute as a temporary workaround (please read the docs first).

**Comment by ginanguy** [comment]
I would like to see if we can get a follow up to this issue. Is there an estimate for when this will be considered to be worked on for remaining GA objects?

**Comment by sfc-gh-asawicki** [comment]
Hey, there is an open community PR (https://github.com/snowflakedb/terraform-provider-snowflake/pull/3641), which is the first step of adding this support to the provider. Adding it is not part of our current priorities, though (you can check them on our [ROADMAP](https://github.com/snowflakedb/terraform-provider-snowflake/blob/main/ROADMAP.md#2025-08-14-roadmap-update)).

**Comment by jamiekt** [comment]
> For now, you can use https://registry.terraform.io/providers/Snowflake-Labs/snowflake/latest/docs/resources/unsafe_execute as a temporary workaround (please read the docs first).

Seems that link has been superseded by https://registry.terraform.io/providers/snowflakedb/snowflake/latest/docs/resources/execute, correct?

**Comment by sfc-gh-jmichalak** [comment]
Yes, that's the correct resource now - `unsafe_execute` has been removed in v1.0.0, see the [migration guide](https://github.com/snowflakedb/terraform-provider-snowflake/blob/main/MIGRATION_GUIDE.md#unsafe_execute-resource-deprecation--new-execute-resource).

**Comment by sbstnkr** [comment]
When can we expect it to be stable?
Git repository resource has been promoted to stable, but if it requires a secret, an api integration needs to be created, therefore creating secret and git repo with one PR is impossible.

**Comment by sfc-gh-jmichalak** [comment]
Hi @sbstnkr 👋 

Thanks for the feedback. It's not on our [current roadmap](https://github.com/snowflakedb/terraform-provider-snowflake/blob/main/ROADMAP.md#2025-12-18-roadmap-update). However, we'll discuss the priorities internally, and we'll update this thread.

**Comment by MickSheppardCorvus** [comment]
Hi,

This is also impacting us. We are using a `snowflake_execute` to workaround it at the moment but would prefer support for the git_https_api type in the api_integration resource.

**Comment by jonathanunderwood** [comment]
To add another voice, please add this to the roadmap.

**Comment by pwhack** [comment]
Does anyone have a sanitized example of the snowflake_execute resource block to create the API integration that can use the Snowflake GitHub App auth? So far all our attempts have failed.

**Comment by MickSheppardCorvus** [comment]
> Does anyone have a sanitized example of the snowflake_execute resource block to create the API integration that can use the Snowflake GitHub App auth? So far all our attempts have failed.

This is what we use:

```hcl
resource "snowflake_execute" "git_api_integration" {
  for_each = local.git_integrations

  execute = <<EOT
CREATE OR REPLACE API INTEGRATION ${each.key}
  API_PROVIDER = git_https_api
  API_ALLOWED_PREFIXES = (${join(",", each.value.allowed_prefixes)})
  API_USER_AUTHENTICATION = (TYPE = SNOWFLAKE_GITHUB_APP)
  ENABLED = TRUE
EOT
  revert  = "DROP API INTEGRATION ${each.key}"
  query   = "DESCRIBE API INTEGRATION ${each.key}"
}

```

**Comment by sfc-gh-jcieslak** [comment]
Hi all,

Yesterday, we released a new provider version with new api integration types: https://github.com/snowflakedb/terraform-provider-snowflake/blob/main/MIGRATION_GUIDE.md#new-featuredeprecation-api-integration-resources-reworked. Please, familiarize yourself with the changes, upgrade, and let us know what do you think. If you feel the feature request was fulfilled, please close the issue.

Thanks 🫡

**Comment by sfc-gh-jcieslak** [comment]
Closing due to long inactivity
